### ISSUE TITLE
[v3] Fix APIs permissions

### ISSUE THREAD

**Issue by florlafuente** [DemocracyOS/democracyos#1579]
We need to fix the endpoints, in order to bring the selected data only to the authorized users and the same with the permisions to post, update or delete data, as checked by @guillecro 
These are the routes wich must be fixed using the middlewares:
- [x] `api/v1.0/users`
- [x] `api/v1.0/settings`
- [x] `api/v1.0/reaction-rule`
- [x] `api/v1.0/reaction-instance`
- [x] `api/v1.0/reaction-vote`
- [x] `api/v1.0/posts`
- [x] `api/v1.0/services/reactions`

**Comment by guillecro** [comment]
The branch will be `fix/api-visibility`

**Comment by guillecro** [comment]
Well, we will have interesting "problems" here

You know, the "app" should also have some kind of role. Like, say for example, the auth functions, like `users/auth/function.js`
Lemme explain you, I have some kind of function that prepares what fields should the querys project or supress. This is an example for listing users
Pay attention, I added a "subject". The idea there is to pass the role. We can discuss the name later.

```
exports.list = function list ({ filter, limit, page, ids, subject }) {
  log.debug('user db-api list')
  let query = {}
  if (filter !== undefined) {
    let filterToJSON = JSON.parse(filter)
    if (filterToJSON.name || filterToJSON.q) {
      filterToJSON.name = { $regex: (filterToJSON.name || filterToJSON.q), $options: 'i' }
      delete filterToJSON.q
    }
    query = filterToJSON
  }
  if (ids) {
    const idsToArray = JSON.parse(ids)
    idsToArray.map((id) => {
      return ObjectId(id)
    })
    query._id = { $in: idsToArray }
  }
  return User
    .paginate(query, { select: allowedFieldsFor(subject), page, limit })
}
```
The, when its time to make the query, with `paginate(...)` or `find().select(...) / findOne().select(...)` 
`allowedFieldsFor(subject)` is called. 
Depending on the role (admin, owner, user, null (anon)) the function adds fields to be visible.

Ok.. guess what happened when i tried to log in.
No email was sended.. Weird. Then I went to the `users/auth/functions.js` and found out that.. woops! The app needs to fetch the data of the user from the data base.. so.. I dont know if this kinds of actions should we make another role like "app"... which.. would have the same priviliges (OR EVEN MORE..) than the admin.

I think this deserves a call with coffee, vainilla cake, and dulce de leche.
Check the branch and see it for yourself!
TTYL!
